atlp-rwanda/atlp-pulse-mobile
# View All Invited Events On Calendar

### What does this PR do?
This PR adds a calendar feature that displays events for each selected date. The calendar includes custom styling for today’s date, the selected date, and event dates, with dots to indicate dates that have events. Users can select a date to view the list of events scheduled on that specific day.

### Description of Task to be completed?

- Integrate a calendar component using react-native-calendars.
- Fetch and display events from the backend based on the selected date.
- Mark dates with dots for days that have events, making it easy for users to identify event days.
- Highlight today’s date and the selected date with specific styles.
- Show the event list when a date with events is selected.

### How should this be manually tested?

1. Clone the repository and navigate to the appropriate branch.
2. Install dependencies by running npm install.
3. Start the application by running npm start.
4. Navigate to the calendar screen and observe:

- Today’s date should be highlighted.
- Dates with events should have a dot indicator.
- Selecting a date should show the events scheduled for that day.

5. Change the month to view events on other dates if available.

### Any background context you want to provide?
This feature is part of the application allowing users to view and track events directly from the calendar. The calendar utilizes the react-native-calendars library, and events are fetched from the backend using GraphQL queries.

### What are the relevant pivotal tracker/Trello stories?
N/A

### Screenshots

![image](https://github.com/user-attachments/assets/a2a98ee9-6f71-4505-94d3-a93f69448f0c)

### Questions
N/A
